File: LeetCode/228. Summary Ranges.js
// https://leetcode.com/problems/summary-ranges/description/

const summaryRanges = (arr) => {
  let start = null;

  const result = [];

  for (let i = 0; i < arr.length; i++) {
    if (start === null) {
      start = i;
    }

    if (arr[i] + 1 === arr[i + 1]) {
      continue;
    }

    if (start === i) {
      result.push(`${arr[i]}`);
    } else {
      result.push(`${arr[start]}->${arr[i]}`);
    }

    start = i + 1;
  }

  return result;
};

console.log(summaryRanges([0, 1, 2, 4, 5, 7])); // ["0->2","4->5","7"]
console.log(summaryRanges([0, 2, 3, 4, 6, 8, 9])); // ["0","2->4","6","8->9"]

File: LeetCode/228. Summary Ranges.py
# https://leetcode.com/problems/summary-ranges/description/


class Solution:
    def summaryRanges(self, nums):
        start = nums[0]

        result = []

        for index in range(len(nums)):
            if index < len(nums) - 1 and nums[index] + 1 == nums[index + 1]:
                continue

            if start == index:
                result.append(f"{nums[index]}")
            else:
                result.append(f"{nums[start]}->{nums[index]}")

            start = index + 1

        return result


solution = Solution()

solution.summaryRanges([0, 1, 2, 4, 5, 7])  # ["0->2","4->5","7"]
solution.summaryRanges([0, 2, 3, 4, 6, 8, 9])  # ["0","2->4","6","8->9"]
